### ISSUE TITLE
[GR-57594] Implement Branch Hinting Proposal for WebAssembly

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9765]
### TL;DR

We plan to implement [the Branch Hinting proposal](https://github.com/WebAssembly/branch-hinting) in GraalWasm.

### Details

With the Branch Hinting proposal, WebAssembly modules can include annotations that indicate whether a particular branch instruction is very likely to take a specific path. This is implemented by introducing a new custom section to the binary format of WebAssembly modules. This new section can reference any `br_if` or `if` instructions in the module's functions and annotate their conditions as being either likely or unlikely to be true. In GraalWasm, we can make use of these annotations to bootstrap our condition profiles and improve warmup.

More details at https://github.com/WebAssembly/branch-hinting/blob/main/proposals/branch-hinting/Overview.md
